Common/Securities/Equity/PatternDayTradingMarginModel.cs

```diff
@@ -0,0 +1,148 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PatternDayTradingMarginModel" />
+        /// </summary>
+        public PatternDayTradingMarginModel()
+            : this(2.0m, 4.0m)
+        {
+        }
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PatternDayTradingMarginModel" />
+        /// </summary>
+        /// <param name="closedmarketleverage"></param>
+        /// <param name="openmarketleverage"></param>
+        public PatternDayTradingMarginModel(decimal closedmarketleverage, decimal openmarketleverage):
+            base(openmarketleverage)
+        {
+            _closedMarketLeverage = closedmarketleverage;
+            _openMarketLeverage = openmarketleverage;   
+        }
+
+        /// <summary>
+        /// Gets the current leverage of the security
+        /// </summary>
+        /// <param name="security">The security to get leverage for</param>
+        /// <returns>The current leverage in the security</returns>
+        public override decimal GetLeverage(Security security)
+        {
+            return 1 / (MaintenanceMarginRequirement * GetMarginCorrection(security));
+        }
+
+        /// <summary>
+        /// Sets the leverage for the applicable securities, i.e, equities
+        /// </summary>
+        /// <remarks>
+        /// Do nothing, we use a constant leverage for this model
+        /// </remarks>
+        /// <param name="security"></param>
+        /// <param name="leverage">The new leverage</param>
+        public override void SetLeverage(Security security, decimal leverage)
+        {
+        }
+
+        /// <summary>
+        /// Gets the total margin required to execute the specified order in units of the account currency including fees
+        /// </summary>
+        /// <remarks>
+        /// If we are in regular market hours, base 4x leverage is used, otherwise leverage is reduced to 2x
+        /// </remarks>
+        /// <param name="security">The security to compute initial margin for</param>
+        /// <param name="order">The order to be executed</param>
+        /// <returns>The total margin in terms of the currency quoted in the order</returns>
+        public override decimal GetInitialMarginRequiredForOrder(Security security, Order order)
+        {
+            var orderFees = security.FeeModel.GetOrderFee(security, order);
+
+            return order.GetValue(security)*InitialMarginRequirement*GetMarginCorrection(security) + orderFees;
+        }
+
+        /// <summary>
+        /// Gets the margin currently alloted to the specified holding
+        /// </summary>
+        /// <param name="security">The security to compute maintenance margin for</param>
+        /// <returns>The maintenance margin required for the </returns>
+        public override decimal GetMaintenanceMargin(Security security)
+        {
+            return base.GetMaintenanceMargin(security)*GetMarginCorrection(security);
```

Comment:
this looks great now, curious why the same pattern isn't used for `GetInitialMarginRequiredForOrder` or `GenerateMarginCallOrder` - it would be nice to minimize the duplicate between this class and the parent class